syntax = "proto3";

package rpc_package;

service GradDescentService{
    rpc grad_descent (server_request) returns (client_reply) {}
}

service GetDataSizeService{
    rpc get_datasize (datasize_request) returns (datasize_reply){}
}


service stop_server{
    rpc stop (stop_request) returns (stop_reply){}
}


message server_request{
    repeated bytes server_grad_para_data=1;
    // repeated int32 server_grad_alpha = 2;
    repeated string server_grad_para_type=2;
    repeated string server_grad_para_shape=3;
}

message client_reply{
    repeated bytes client_grad_para_data=1;
    // repeated int32 grad_alpha=2;
    repeated string client_grad_para_type=2;
    repeated string client_grad_para_shape=3;
    // repeated int32 datasize=2;
}

message datasize_request{
    int32 size=1;
}

message datasize_reply{
    int32 size=1;
}


message stop_request{
    string message=1;
}

message stop_reply{
    string message=1;
}